src/Diagnostics.RuntimeHost/Services/DiagnosticTranslator/DiagnosticTranslatorService.cs

```diff
@@ -0,0 +1,320 @@
+﻿using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Diagnostics.RuntimeHost.Services.CacheService;
+using System.Net.Http.Headers;
+using Diagnostics.ModelsAndUtils.Models;
+using System.Data;
+using Microsoft.CodeAnalysis;
+using Diagnostics.ModelsAndUtils.Attributes;
+
+namespace Diagnostics.RuntimeHost.Services.DiagnosticsTranslator
+{
+    public interface IDiagnosticTranslatorService
+    {
+        Task<Response> GetResponseTranslations(Response response, string language);
+        Task<IEnumerable<DiagnosticApiResponse>> GetMetadataTranslations(IEnumerable<DiagnosticApiResponse> listDetectorsResponse, string language);
+    }
+
+    public class DiagnosticTranslatorService : IDiagnosticTranslatorService
+    {
+        private IConfiguration _config;
+        private static string _enableLocalization;
+        private static string _translatorSubscriptionKey;
+        private static string _translatorBaseURL;
+        private static string _translatorApiURL;
+        private static string _apiVersion;
+
+        private ITranslationCacheService _translationCacheService;
+        private static readonly HttpClient _httpClient = new HttpClient();
+
+        private void InitializeHttpClient()
+        {
+            _httpClient.Timeout = TimeSpan.FromSeconds(10);
+            _httpClient.BaseAddress = new Uri(_translatorBaseURL);
+            _httpClient.DefaultRequestHeaders.Accept.Clear();
+            _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            _httpClient.DefaultRequestHeaders.Add("Ocp-Apim-Subscription-Key", _translatorSubscriptionKey);
+        }
+
+        /// <summary>
+        /// Creates a new instance of DiagnosticTranslatorService
+        /// </summary>
+        /// <param name="configuration">Configuration object</param>
+        /// <param name="translationCacheService">Translation cache service</param>
+        public DiagnosticTranslatorService(IConfiguration configuration, ITranslationCacheService translationCacheService)
+        {
+            _config = configuration;
+            _enableLocalization = _config[$"DiagnosticTranslator:EnableLocalization"];
```

Comment:
Can you rename `DiagnosticTranslator:EnableLocalization` to `DiagnosticTranslator:Enabled`. Just to keep consistency with what we called the feature flag for other services.

Reply:
Updated